partiql-ast/src/main/java/org/partiql/ast/AstNode.java

```diff
@@ -1,20 +1,22 @@
 package org.partiql.ast;
 
+import lombok.Getter;
+import lombok.Setter;
 import org.jetbrains.annotations.NotNull;
 
-import java.util.Collection;
-import java.util.Random;
+import java.util.List;
 
 /**
  * TODO docs, equals, hashcode
  * TODO support source location -- https://github.com/partiql/partiql-lang-kotlin/issues/1608
  */
+@Setter
+@Getter
 public abstract class AstNode {
-    @NotNull
-    public String tag = "Ast-" + String.format("%06x", new Random().nextInt());
+    private int tag = 0;
```

Comment:
Let's add an explicit getter/setter like in the plans.

Reply:
Updated